charts/lily/templates/statefulset.yaml

```diff
@@ -57,8 +59,37 @@ spec:
 
             # import snapshot if enabled
             {{- if .Values.daemon.importSnapshot.enabled }}
-              {{- include "sentinel-lily.chainImportArgs" . | indent 10 }}
-            {{- else }}
+            if [ -f "/var/lib/lily/datastore/_imported" ]; then
+              echo "Skipping import, found /var/lib/lily/datastore/_imported file."
+              echo "Ensuring secrets have correct permissions."
+              chmod 0600 /var/lib/lily/keystore/*
+              exit 0
+            fi
+
+            echo "*** Downloading snapshot from url {{ .Values.daemon.importSnapshot.url }}..."
+            curl -sL -o ./aria2-1.36.0.deb https://github.com/q3aql/aria2-static-builds/releases/download/v1.36.0/aria2-1.36.0-linux-gnu-64bit-build1.deb
+            dpkg -i ./aria2-1.36.0.deb
+            rm ./aria2-1.36.0.deb
+
+            (cd /var/lib/lily/datastore && time aria2c -x16 -k1M -o snapshot.car {{ .Values.daemon.importSnapshot.url }}
+            status=$?
+            if [ $status -ne 0 ]; then
+              rm /var/lib/lily/datastore/snapshot.car
+              echo "...download failed: $status"
+              exit $status
+            fi
+
+            echo "*** Importing snapshot..."
+            lily init --import-snapshot=/var/lib/lily/datastore/snapshot.car
+            status=$?
+            rm /var/lib/lily/datastore/snapshot.car
```

Comment:
would it be worth leaving the snapshot around in the event the import fails?

Reply:
🤝 Yeah, that'd be better.